test/unit/lodash.spec.js

```diff
@@ -0,0 +1,583 @@
+'use strict';
+
+var chai = require('chai');
+var expect = chai.expect;
+var _ = require('lodash');
+var datasets = require('../../app/datasets');
+
+describe('Lodash training', function() {
+    var params;
+
+    describe('functions', function() {
+        beforeEach(function() {
+            params = datasets.functions();
+        });
+        it('should match types of passing elements', function() {
+            var obj = params[0];
+            expect(obj).to.be.an('object');
+            expect(params).to.have.length(1);
+        });
+        it('should contain only one property square', function() {
+            var obj = params[0];
+            var i = 0;
+            for (var key in obj) {
+                if (obj.hasOwnProperty(key) && key === 'square') {
+                    if (obj[key] instanceof Function) {
+                        i++;
+                    }
+                }
+            }
+            expect(i).to.eql(1);
+        });
+        it('should not contain own property cube', function() {
+            var obj = params[1];
+            for (var key in obj) {
+                if (obj.hasOwnProperty(key)) {
+                    expect(key).to.not.eql('cube');
+                }
+            }
+        });
+        it('should return array property names of functions', function() {
+            expect(_.functions.apply(_, params)).to.eql(['square']);
+        });
+    });
+
+    describe('functionsIn', function() {
+        beforeEach(function() {
+            params = datasets.functionsIn();
+        });
+        it('should match types of passing elements', function() {
+            var element = params[0];
+            expect(element).to.be.an('object');
+            expect(params).to.have.length(1);
+        });
+        it('should contain only one property square', function() {
+            var obj = params[0];
+            var i = 0;
+            for (var key in obj) {
+                if (obj.hasOwnProperty(key)) {
+                    if (obj[key] instanceof Function) {
+                        i++;
+                    }
+                }
+            }
+            expect(i).to.eql(1);
+        });
+        it('should not contain own property cube', function() {
+            var obj = params[0];
+            expect(obj).to.have.property('cube');
+            expect(obj.cube).to.be.an('function');
+        });
+        it('should return array property names of functions', function() {
+            expect(_.functionsIn.apply(_, params)).to.eql(['square', 'cube']);
+        });
+    });
+
+    describe('pick', function() {
+        describe('pick1', function() {
+            beforeEach(function() {
+                params = datasets.pick1({
+                    'red': 1,
+                    'green': 2,
+                    'blue': 3,
+                    'yellow': 4
+                });
+            });
+            it('should match types of passing elements', function() {
```

Comment:
Okay, so by looking on this test I am certainly well informed... Well even to be honest I am able to just write everything without any lodash knowledge. This is wrong, because as I said before you need to ensure that user is not able to bypass test cases, but also not to be able to "copy" answer from them!